secret_store/src/key_server_cluster/io/deadline.rs

```diff
@@ -52,9 +58,9 @@ impl<F, T> Future for Deadline<F> where F: Future<Item = T, Error = io::Error> {
 
 	fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
 		match self.future.poll() {
-			Ok(Async::Ready((result, _other))) => Ok(Async::Ready(result)),
+			Ok(Async::Ready(result)) => Ok(Async::Ready(result)),
```

Comment:
`self.future.poll()` will be enough here

Reply:
Probably worth removing `Deadline` at all and replace it with `Timeout` (not a part of this PR, though)